Feature #7652 - CoreOS support

endyman opened the pull request:
Implements foreman CoreOS OperatingSystem Support (http://projects.theforeman.org/issues/7652). PXE and cloud-config Templates will be send in a different PR to the community-template repository.

Commit:
fixes #7652 - Implements CoreOS OS support

fixed OS test case for CoreOS

Implements feature-7652

fixed OS test case for CoreOS

added free coreOS icon - kudos lzap

readded official CoreOS logo

Added testcases for coreos and use boot_files_uri method from parent

endyman commented:
Added PR to community repo: https://github.com/theforeman/community-templates/pull/124

mmoll commented:
[test]

mmoll commented:
Tests failed, that's an easy one, see Jenkins output.

dLobatog commented:
[test]

dLobatog commented:
@endyman I'm going to try it right now, thanks! Just one comment, you should `git commit --amend` this to comply with [this format](http://projects.theforeman.org/projects/foreman/wiki/Reviewing_patches-commit_message_format)

dLobatog commented:
[test]

endyman commented:
If there is any License issue with the logo we can go with the "dummy" icons generated by @lzap  i.e. https://github.com/lzap/foreman-dummy-icons/blob/master/21x21-deepskyblue/c.png

lzap commented:
I was unable to find a page with copyright and licensing info for their logo. Thus, yes it's a license issue and we need the dummy one. You can try to contact the CoreOS team if you want to. I gave the googling only one minute....

maal commented:
I will ask the coreos team if they could provide one for us.

endyman commented:
I removed the logo and replaced it by a "free" version from @lzap until we have feedback.
Here is the request on the core os mailing-list from @maal 
https://groups.google.com/forum/#!topic/coreos-user/SEW9prP_nb0

dLobatog commented:
[test]

robszumski commented:
Hi everyone, I'm the designer at CoreOS. What type of sign off are you looking for on the copyright?

If I provide you with this image, is that ok?

![image](https://cloud.githubusercontent.com/assets/755540/4929821/60b1cc72-6559-11e4-949f-0f5740f4c5f9.png)

dLobatog commented:
@robszumski thanks for chiming in!

 We found that image in your press-kit, but we were a bit concerned there were no terms or use, license or anything similar, so we weren't sure whether these logos could be use in other programs. 

Is there any official policy on that? If not, I guess we can use that icon logo without consequences but I'd humbly recommend adding a license to the press-kit or in your web somewhere, something along the lines of [this](https://www.docker.com/legal/marks_and_logos/). 

Using this kind of content without a license is copyright infringement by default, and _fair use_ is not a legal doctrine all around the world. 

Sorry in advance in case we're going overboard about this...

robszumski commented:
I added a license to the Press Kit, but you can also consider this permission to use the globe logo.

dLobatog commented:
[test]

ohadlevy commented on app/models/operatingsystems/coreos.rb:
```diff
@@ -0,0 +1,43 @@
+
+  def boot_files_uri(medium, architecture, host = nil)
+    raise ::Foreman::Exception.new(N_("invalid medium for %s"), to_s) unless media.include?(medium)
+    raise ::Foreman::Exception.new(N_("invalid architecture for %s"), to_s) unless architectures.include?(architecture)
```
since this method already exists in the parent class, cant you do something like

``` ruby
def boot_files_uri(..)
 super(..).gsub.. ?
end
```

ohadlevy replied:
the same is probably true for debian...

johscheuer replied:
I will have a look. Thanks for the hint.

endyman replied:
@ohadlevy should fix this for debian in this PR too? I would suggest to create a second PR for debian.

what do you think?

ohadlevy replied:
> yeah, it make sense, probably to merge it prior to this PR.. thanks!

orrabin commented:
[test]

dLobatog commented:
@endyman :bow: Really, really sorry for the delay. I've finally got around testing this and :+1: . I will change the FactoryGirl.create calls (they hit the db) to FactoryGirl.build calls on merge but that's it.

dLobatog commented:
Merged as 5a594178a850de6e71e0b11e4a3de8b8159e5a79, thanks @endyman , and @robszumski for the logo!
